app/controllers/DraftMetadataChecksResultsController.scala

```diff
@@ -33,16 +34,73 @@ class DraftMetadataChecksResultsController @Inject() (
       for {
         reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
         consignmentStatuses <- consignmentStatusService.getConsignmentStatuses(consignmentId, token)
+        errorType <- if (errorTypeDeterminedFromFile(consignmentStatuses)) draftMetadataService.getErrorType(consignmentId) else Future.successful(FileError.UNKNOWN)
```

Comment:
Just a suggestion, you may filter the `DraftMetadata` status and check the value here: 
```
draftMetadataStatus = consignmentStatuses.find(_.statusType == DraftMetadataType.id).map(_.value).getOrElse("")
        errorType <- if (draftMetadataStatus != CompletedValue.value) draftMetadataService.getErrorType(consignmentId) else Future.successful(FileError.UNKNOWN)
```

and then use the status to check the `draftMetadataProgress` for the HTML pages
```

val resultsPage = {
          // leaving original page for no errors
          if (draftMetadataStatus == CompletedValue.value) {
            views.html.draftmetadata.draftMetadataChecksResults(consignmentId, reference, getDraftMetadataProgress(draftMetadataStatus), request.token.name)
          } else {
            if (isErrorReportAvailable(errorType)) {
              views.html.draftmetadata
                .draftMetadataChecksWithErrorDownload(
                  consignmentId,
                  reference,
                  getDraftMetadataProgress(draftMetadataStatus),
                  request.token.name,
                  actionMessage(errorType),
                  detailsMessage(errorType)
                )
            } else {
              views.html.draftmetadata
                .draftMetadataChecksErrorsNoDownload(
                  consignmentId,
                  reference,
                  getDraftMetadataProgress(draftMetadataStatus),
                  request.token.name,
                  actionMessage(errorType),
                  detailsMessage(errorType)
                )
            }
          }
        }
```

Reply:
refactored to evaluate errorType in separate def
this allows checking failed as well as completed
then use error type to determine if existing success page should be used